lib/archive_importer/entity_importer.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class ArchiveImporter
+  class EntityImporter
+    include ArchiveValidator::EntitiesHelper
+    include Diaspora::Logging
+
+    def initialize(json, user)
+      @json = json
+      @user = user
+    end
+
+    def import
+      self.persisted_object = Diaspora::Federation::Receive.perform(entity)
+    rescue DiasporaFederation::Entities::Signable::SignatureVerificationFailed => e
+      logger.warn "#{self}: #{e}"
+    rescue DiasporaFederation::Discovery::InvalidDocument,
+           DiasporaFederation::Discovery::DiscoveryError,
+           ActiveRecord::RecordInvalid
+    end
```

Comment:
I think a comment in here why we want to ignore these if they happen (and not even log) would be nice :)

Reply:
Well, maybe we actually should log this exceptions here. I just ran my tests once again and I couldn't find a valid case when it's totally fine to ignore the exception. Maybe we can hide some warnings later, but it is better to be more verbose at the beginning. What do you think?

Reply:
`DiasporaFederation::Discovery::InvalidDocument` happened here once when a pod replied `500` on any request (should be logged imo).
`DiasporaFederation::Discovery::DiscoveryError` happens when relayable author is not fetchable (should be logged)
`ActiveRecord::RecordInvalid` doesn't happen in real life at all ATM (only in tests). Probably should be removed then?

Reply:
> `DiasporaFederation::Discovery::InvalidDocument` happened here once when a pod replied 500 on any request (should be logged imo).
> `DiasporaFederation::Discovery::DiscoveryError` happens when relayable author is not fetchable (should be logged)

Sounds good to me.

> `ActiveRecord::RecordInvalid` doesn't happen in real life at all ATM (only in tests). Probably should be removed then?

Honestly, I trust your judgement better than mine on this one :) However my intuition would be to be defensive in case a third-party implementation throws something weird at us.